applications/nrf_desktop/doc/dfu.rst

```diff
@@ -0,0 +1,129 @@
++====================+========================+=============+=========================+==================+
+| ``hids``           | ``config_event``       | ``dfu``     |                         |                  |
++                    +                        |             |                         |                  |
+| ``usb_state``      |                        |             |                         |                  |
++--------------------+------------------------+             |                         |                  |
+| ``hids``           | ``config_fetch_event`` |             |                         |                  |
++                    +                        |             |                         |                  |
+| ``usb_state``      |                        |             |                         |                  |
++--------------------+------------------------+             |                         |                  |
+| ``hid_state``      | ``hid_report_event``   |             |                         |                  |
++--------------------+------------------------+-------------+-------------------------+------------------+
+
+
+Configuration
+*************
+
+The DFU module is selected using ``CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_ENABLE``.
+Because it uses the :ref:`nrf_desktop_config_channel` for the transmission of the update image, it requires the transport option ``CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE`` to be selected.
+
+When the update image is being received, it must stored in the dedicated partition on the flash memory.
+For more information about configuring the memory layout in the application, see :ref:`nrf_desktop_flash_memory_layout`.
+
+
+Implementation details
+**********************
+
+The DFU module implementation is centered around transmitting and receiving update images and can be broken down into the following stages:
+
+* `Protocol operations`_ - How the module exchanges information with the host.
+* `Partition preparation`_ - How the module prepares for receiving an image.
+
+Protocol operations
+===================
+
+The DFU module depends on the :ref:`nrf_desktop_config_channel` for the correct and secure data transmission between the host and the device.
+The module provides a simple protocol that allows an update tool on the host to pass the update image to the device.
+
+The following commands are available for controlling the module operation:
+
+* `fwinfo`_ - Pass the information about currently executed image from the device to the host.
+* `reboot`_ - Reboot the device.
+* `start`_ - Start the new update image transmission.
+* `data`_ - Pass a chunk with the update image data from the device to the host.
+* `sync`_ - Check the progress of the update image transmission.
+
+fwinfo
+------
+
+The ``fwinfo`` command is issued by sending a ``config_fetch_event`` from the host tool through the DFU module, with the ``fwinfo`` option for identification.
```

Comment:
I think we do not need to describe this here, since its common for all the config_channel listeners.
And `config_fetch_event `is actually sent by the listener as reaction for ``config_fetch_request_event``

Reply:
I think we should only describe:
* if the option can be set or get (or both)
* what data is associated with the option (as it is done here).

Informing about implementation details is unnecessary with the current implementation.

Reply:
We don't treat the options normally here and this is why I prefer to keep it. Options work as a remote commands for DFU. If anything we should probably rename config channel 'options' to something more universal. But let's discuss that later during the polishing review.